tests/infer/test_inference.py

```diff
@@ -745,3 +745,33 @@ def model():
         loss = svi.step()
         if step % 20 == 0:
             logger.info("step {} loss = {:0.4g}".format(step, loss))
+
+
+@pytest.mark.stage("integration", "integration_batch_1")
+def test_sequential_plating_sum():
+    """Example from https://github.com/pyro-ppl/pyro/issues/2361"""
+
+    def model(data):
+        x = pyro.sample('x', dist.Bernoulli(torch.tensor(0.5)))
+        for i in pyro.plate('data_plate', len(data)):
+            pyro.sample('data_{:d}'.format(i),
+                        dist.Normal(x, scale=torch.tensor(0.1)),
+                        obs=data[i])
+
+    def guide(data):
+        p = pyro.param('p', torch.tensor(0.5))
+        pyro.sample('x', pyro.distributions.Bernoulli(p))
+
+    data = torch.cat([torch.randn([5]), 1. + torch.randn([5])])
+    adam = optim.Adam({"lr": 0.01})
+    loss_fn = RenyiELBO(alpha=0, num_particles=30, vectorize_particles=True)
+    svi = SVI(model, guide, adam, loss_fn)
+
+    for step in range(1001):
+        loss = svi.step(data)
+        if step % 20 == 0:
+            logger.info("step {} loss = {:0.4g}".format(step, loss))
+
+    expected_p = torch.tensor(0.5)
+    actual_p = pyro.param("p").detach()
+    assert_close(actual_p, expected_p, atol=0.05)
```

Comment:
It is surprising to me that without enumeration, SVI works for this pair of `model`/`guide`. Bernoulli is a discrete distribution, so grad can't pass through its samples. What happens if you change the initial parameter `p` to something different from `0.5`, for example `p = pyro.param('p', torch.tensor(0.2))`?

If SVI fails, I would suggest to only run 1 step `svi.step(data)` to verify `RenyiELBO` can run with a sequential plate. Alternatively, I guess you can change `Bernoulli` to some other continuous distribution, for example `dist.Normal(0, 1)` in `model` and `dist.Normal(p, 0.1)` in the guide, to make the inference work.

Reply:
Ah, I didn't actually do any of those checks---just put in a dummy check as we didn't care about the answer in this case.
I thought the use of `score_parts` within `RenyiElbo` handles the discrete Bernoulli?

In any case, I'll just run this for one step as we just need to be sure it 'runs'. :)

Reply:
You are right! `score_parts` might remedy that issue. Thanks for reminding me about it!